pkg/drivers/kvm/domain.go

```diff
@@ -0,0 +1,124 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kvm
+
+import (
+	"bytes"
+	"fmt"
+	"text/template"
+
+	libvirt "github.com/libvirt/libvirt-go"
+	"github.com/pkg/errors"
+)
+
+const domainTmpl = `
+<domain type='kvm'>
+  <name>{{.MachineName}}</name> 
+  <memory unit='MB'>{{.Memory}}</memory>
+  <vcpu>{{.CPU}}</vcpu>
+  <features>
+    <acpi/>
+    <apic/>
+    <pae/>
+  </features>
+  <os>
+    <type>hvm</type>
+    <boot dev='cdrom'/>
+    <boot dev='hd'/>
+    <bootmenu enable='no'/>
+  </os>
+  <devices>
+    <disk type='file' device='cdrom'>
+      <source file='{{.ISO}}'/>
+      <target dev='hdc' bus='ide'/>
+      <readonly/>
+    </disk>
+    <disk type='file' device='disk'>
+      <driver name='qemu' type='raw' cache='default' io='threads' />
+      <source file='{{.DiskPath}}'/>
+      <target dev='hda' bus='ide'/>
+    </disk>
+    <interface type='network'>
+      <source network='{{.Network}}'/>
+    </interface>
+    <interface type='network'>
+      <source network='{{.PrivateNetwork}}'/>
+    </interface>
+    <serial type='pty'>
+      <source path='/dev/pts/2'/>
+      <target port='0'/>
+    </serial>
+    <console type='pty' tty='/dev/pts/2'>
+      <source path='/dev/pts/2'/>
+      <target port='0'/>
+    </console>
+  </devices>
+</domain>
+`
+
+func (d *Driver) getDomain() (*libvirt.Domain, *libvirt.Connect, error) {
+	conn, err := getConnection()
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "getting domain")
+	}
+
+	dom, err := conn.LookupDomainByName(d.MachineName)
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "looking up domain")
+	}
+
+	return dom, conn, nil
+}
+
+func getConnection() (*libvirt.Connect, error) {
+	conn, err := libvirt.NewConnect(qemusystem)
+	if err != nil {
+		return nil, errors.Wrap(err, "Error connecting to libvirt socket")
```

Comment:
We've had a bunch of bugs about users not understanding this or a similar error. Could we make this a bit more helpful, with some suggestions about how to resolve it?